woltka/cli.py

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/env python3
+
+# ----------------------------------------------------------------------------
+# Copyright (c) 2020--, Qiyun Zhu.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+"""Standalone command-line interface (CLI) of the program.
+"""
+
+import click
+
+from . import __version__
+from .workflow import workflow
+
+
+class NatOrder(click.Group):
+    """Natural ordering of click command groups.
+    """
+    def list_commands(self, ctx):
+        return self.commands.keys()
+
+
+@click.version_option(__version__)
+@click.group(cls=NatOrder)
+def cli():
+    pass
+
+
+# `gotu` is a simplified wrapper of `classify`
+
+@cli.command('gotu')
+@click.option(
+    '--input', '-i', 'input_path', required=True,
+    type=click.Path(exists=True, file_okay=True, dir_okay=True),
+    help=('Path to a multiplexed alignment file or a directory of per-sample '
+          'alignment files.'))
+@click.option(
+    '--output', '-o', 'output_path', required=True,
+    type=click.Path(writable=True),
+    help=('Path to output gOTU table.'))
+@click.option(
+    '--ambig/--no-ambig', default=True,
+    help=('Allow one sequence to be assigned to multiple gOTUs. Each hit '
+          'will be counted as 1 / k (k is the totally number of hits).'))
+@click.option(
+    '--map', '-m', 'map_fps', type=click.Path(exists=True), multiple=True,
+    help=('Map of nucleotides to genomes.'))
+@click.pass_context
+def gotu(ctx, **kwargs):
+    """Generate a gOTU table based on sequence alignments.
+    """
+    ctx.invoke(classify, **kwargs)
```

Comment:
Is there a reason for invoking classify here, can you not just call workflow directly?

Reply:
The reason is that invoking `classify` will inherit all the default parameters defined in the click decorators of `classify`. If one directly calls `workflow`, these defaults are gone.